src/drivers/intel/cavs/hda-dma.c

```diff
@@ -229,21 +229,20 @@ static void hda_dma_get_dbg_vals(struct hda_chan_data *chan,
 		struct hda_dbg_data *dbg_data = &(chan)->dbg_data; \
 		if (dbg_data->cur_sample < HDA_DMA_PTR_DBG_NUM_CP) { \
 			uint32_t bne = \
-				dbg_data->last_bne[HDA_DBG_PRE] << 4 | \
-				dbg_data->last_bne[HDA_DBG_POST]; \
+				((uint32_t)dbg_data->last_bne[HDA_DBG_PRE] << \
+				 4) | dbg_data->last_bne[HDA_DBG_POST]; \
 			uint32_t info = ((chan)->direction << 16) | bne; \
 			uint32_t wp = \
-				(dbg_data->last_wp[HDA_DBG_PRE] << 16) | \
-				(dbg_data->last_wp[HDA_DBG_POST] & 0xFFFF); \
+				((uint32_t)dbg_data->last_wp[HDA_DBG_PRE] << \
+				 16) | dbg_data->last_wp[HDA_DBG_POST]; \
 			uint32_t rp = \
-				(dbg_data->last_rp[HDA_DBG_PRE] << 16) | \
-				(dbg_data->last_rp[HDA_DBG_POST] & 0xFFFF); \
+				((uint32_t)dbg_data->last_rp[HDA_DBG_PRE] << \
+				 16) | dbg_data->last_rp[HDA_DBG_POST]; \
```

Comment:
Can we have some macros for these magic numbers to match the comments explaining the data format ?

Reply:
I've decided to create generic macros for such operations